src/cloud-api-adaptor/test/e2e/common_suite.go

```diff
@@ -499,6 +500,41 @@ func DoTestPodsMTLSCommunication(t *testing.T, e env.Environment, assert CloudAs
 
 }
 
+func DoTestImageDecryption(t *testing.T, e env.Environment, assert CloudAssert, kbs *pv.KeyBrokerService) {
+	image := "ghcr.io/confidential-containers/cloud-api-adaptor/nginx-encrypted:20240123"
+	var kbsEndpoint string
+	if ep := os.Getenv("KBS_ENDPOINT"); ep != "" {
+		kbsEndpoint = ep
+	} else if kbs == nil {
+		t.Skip("Skipping because KBS config is missing")
+	} else {
+		// skopeo inspect \
+		//   docker://ghcr.io/confidential-containers/cloud-api-adaptor/nginx-encrypted:20240123 \
+		//   | jq .Labels
+		// {
+		//   "coco-key-b64": "pHSE5N+T/3GGfb/umaWgB8bfHc/dQWvmxdsjoWam0Vs=",
+		//   "coco-key-id": "default/key/nginx-encrypted",
+		// }
+		keyID := "default/key/nginx-encrypted"
+		key := []byte{
+			164, 116, 132, 228, 223, 147, 255, 113, 134, 125,
+			191, 238, 153, 165, 160, 7, 198, 223, 29, 207,
+			221, 65, 107, 230, 197, 219, 35, 161, 102, 166,
+			209, 91}
+
+		_ = kbs.SetImageDecryptionKey(keyID, key)
+		_ = kbs.EnableKbsCustomizedResourcePolicy("allow_all.rego")
+		_ = kbs.EnableKbsCustomizedAttestationPolicy("allow_all.rego")
+		kbsEndpoint, _ = kbs.GetCachedKbsEndpoint()
```

Comment:
I'm happy to update this in #2055 (if I remember), but in ee76f06936c2df02226230e04b82bd5b3fefc83b I added checks of the errors thrown as I've seen at least one example of where we had issues and ignored them which made debugging a lot harder in the CI, so maybe we should consider that here too even if it is more golang boilerplate?

Reply:
sure, just for this test?

Reply:
Yeah, that's fine. I've handled the rest elsewhere.